Client/Frontend/Browser/TopTabsViewController.swift

```diff
@@ -337,6 +330,18 @@ extension TopTabsViewController: TabSelectionDelegate {
     }
 }
 
+extension TopTabsViewController: TabEventHandler {
+    func tab(_ tab: Tab, didLoadFavicon favicon: Favicon?, with: Data?) {
+        // Make sure animations only happen on the main thread.
+        DispatchQueue.main.async {
```

Comment:
We should already be on `main`, so perhaps just `assert` that rather than dispatch another block.